fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/ReviewQualityCheckSettingsCard.kt

```diff
@@ -0,0 +1,117 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.shopping.ui
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import org.mozilla.fenix.R
+import org.mozilla.fenix.compose.SwitchWithLabel
+import org.mozilla.fenix.compose.annotation.LightDarkPreview
+import org.mozilla.fenix.compose.button.SecondaryButton
+import org.mozilla.fenix.theme.FirefoxTheme
+
+/**
+ * Review quality check settings card UI. Contains toggles to disable product recommendations and
+ * the entire review quality check feature.
+ *
+ * @param productRecommendationsEnabled The current state of the product recommendations toggle.
+ * @param onProductRecommendationsEnabledStateChange Invoked when the user changes the product
+ * recommendations toggle state.
+ * @param onTurnOffReviewQualityCheckClick Invoked when the user opts out of the review quality check feature.
+ * @param modifier Modifier to apply to the layout.
+ */
+@Composable
+fun ReviewQualityCheckSettingsCard(
+    productRecommendationsEnabled: Boolean,
+    onProductRecommendationsEnabledStateChange: (Boolean) -> Unit,
+    onTurnOffReviewQualityCheckClick: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    ReviewQualityCheckExpandableCard(
+        modifier = modifier,
+        title = stringResource(R.string.review_quality_check_settings_title),
+    ) {
+        SettingsContent(
+            productRecommendationsEnabled = productRecommendationsEnabled,
+            onProductRecommendationsEnabledStateChange = onProductRecommendationsEnabledStateChange,
+            onTurnOffReviewQualityCheckClick = onTurnOffReviewQualityCheckClick,
+            modifier = Modifier.fillMaxWidth(),
+        )
+    }
+}
+
+@Composable
+private fun SettingsContent(
```

Comment:
Not sure we need an extra function indirection with the same shape/params, curious what was the thought process to add this? Also, removing the need for an extra preview.

Reply:
Ahh gotcha, makes sense! thanks for the explanation!